build: update dependencies, lint and clean project, add ci

Quentin-Guillemin opened the pull request.

Quentin-Guillemin committed:
style: remove comments and unnecessary public

Quentin-Guillemin committed:
fix: update spring boot and dependencies

Quentin-Guillemin committed:
fix: rollback to java 11

Quentin-Guillemin committed:
style: rearrange files

Add LICENCE and NOTICE

Add maven-license-plugin and notice-maven-plugin

Quentin-Guillemin committed:
fix(SecurityConfiguration)!: update deprecated code

Quentin-Guillemin committed:
feat: adjust cache duration in application.yml

Quentin-Guillemin committed:
fix: remove log

Quentin-Guillemin committed:
refactor: lint some code

Quentin-Guillemin committed:
refactor: lint some code

Quentin-Guillemin committed:
feat: addnvm commands

jgribonvald committed:
feat: update ci test to check on new branch, license and notice

jgribonvald committed:
fix: generated NOTICE

jgribonvald committed:
chore(deps): update maven wrapper

jgribonvald committed:
style: apply license headers on source files

jgribonvald committed:
doc: add doc link to process maven wrapper update

Quentin-Guillemin committed:
chore: update spring boot and uPortal-soffit-renderer

Quentin-Guillemin committed:
build: update spring boot

Quentin-Guillemin committed:
ci: add dependabot

Quentin-Guillemin committed:
ci: add husky

Quentin-Guillemin committed:
ci: disable build and test

Quentin-Guillemin committed:
build(deps): bump spring-boot-starter-parent from 2.7.11 to 2.7.15

Quentin-Guillemin committed:
refactor: replace @Values with bean properties

Quentin-Guillemin committed:
refactor: clean and lint code

Quentin-Guillemin committed:
build(deps): bump

* spring-boot from 2.7.15 to 2.7.16

* hazelcast from 5.2.1 to 5.3.2

* jakarta.xml.bind-api from 4.0.0 to 4.0.1

* license-maven-plugin from 1.10.b1 to 4.3

Quentin-Guillemin committed:
feat: add profiles

Quentin-Guillemin committed:
fix: scm

Quentin-Guillemin committed:
fix(conf): use list of properties

Quentin-Guillemin committed:
build: change packaging from jar to war

Quentin-Guillemin committed:
build: fix war error

EwannRoux committed:
fix(Ressources-Diffusables-API): add missing check to isEmpty of filter

EwannRoux committed:
feat(Ressources-Diffusables-API): rewrite json downloading logic when expired, now use a temp file before overwriting previous, and modify some getters and setters for future spying and mocking in futures tests

EwannRoux committed:
feat(Ressources-Diffusables-API): add mereFamille and membreFamille to RessourceDiffusable

EwannRoux committed:
feat(Ressources-Diffusables-API): replace JSON manual parsing in lireFichier by ObjectMapper and objects mirroring json structure, with factory constructor to RessourceDiffusable

EwannRoux committed:
fix(Ressources-Diffusables-API): correct telechargerFichier way of downloading json file, now with a relative URL set in application.yml

EwannRoux committed:
feat(Ressources-Diffusables-API): remove all ehcache usages from project

jgribonvald commented on doc/dev.md:
```diff
@@ -0,0 +1,19 @@
+`./mvnw license:format` add the license header when missing. If a header is existing, it is updated to the new one.
+`./mvnw license:remove` remove existing license header
+
+## In dev process when changing the entities
```
attention copy paste abusif. Il n'y a pas ce type de fonctionnement dans ce projet.

EwannRoux committed:
refactor(Ressources-Diffusables-API): move downloading and parsing of json in two new services, rewrite parsing service and models in order to use Jackson annotations instead of mirroring objects, and detect unknowns properties

EwannRoux committed:
refactor(Ressources-Diffusables-API): remove now unused mirroring object from previous json method

EwannRoux committed:
chore(Ressources-Diffusables-API): update springboot version to 2.7.18

EwannRoux committed:
chore: add *application-*.yml to war excludes

EwannRoux committed:
chore: remove externalized config from application.yml

EwannRoux committed:
fix: add missing properties to AppProperties

EwannRoux committed:
chore: update notice (removed ehcache since no longer used)

EwannRoux committed:
chore: update dev.md

EwannRoux committed:
chore: remove beans.xml

EwannRoux committed:
feat: remove unused cacheDuration from GARProperties

EwannRoux committed:
feat: remove unused size and pageCount from controller

EwannRoux committed:
feat: add pagination object to response, and use ResponseEntity in controller

EwannRoux committed:
feat: remove unused import resource in Application

EwannRoux committed:
feat: create reworked service interfaces

EwannRoux committed:
feat: add service configurations

EwannRoux committed:
feat: add nogar properties

EwannRoux committed:
feat: create utils class

EwannRoux committed:
feat: create tuple class

EwannRoux committed:
feat: add log to exception handlers

EwannRoux committed:
test: comment controller test class during service rewriting

EwannRoux committed:
feat: update RessourceDiffusable constructor

EwannRoux committed:
feat: add downloadFileName properties to GARProperties

EwannRoux committed:
feat: add service implementations

EwannRoux committed:
test: add PaginationRequestMatcher and RessourceDiffusableFilterMatcher

EwannRoux committed:
refactor: add lombock, make field final and merge constructors in ApiResponse

EwannRoux committed:
feat: replace nullable boolean used in Filter by tri-values enum and  enum wrappers

EwannRoux committed:
test: adapt ApiRessourcesDiffusablesControllerTest to new endpoints and mock new services

EwannRoux committed:
test: create class that generate a list of RessourceDiffusable with random values

EwannRoux committed:
test: add test for ParserService and its Jackson implementation

EwannRoux committed:
refactor: rename FilterBoolean value IGNORED to UNKNOWN for better clarity

EwannRoux committed:
fix: fix filter never returning false and improve FilterBooleanWrapper

remove usage of nullable boolean from RessourceDiffusableFilter for clarity, using FilterBoolean and FilterBooleanWrapper instead and add required methods to FilterBooleanWrapper

EwannRoux committed:
fix: fix uri construction of local file for both implementation of DAO and put examples in conf

EwannRoux committed:
chore: add conf example

EwannRoux committed:
style: remove unreachable unused code

EwannRoux committed:
style: move ClockConfiguration to correct folder

EwannRoux committed:
test: temporarily comment out all test from  RessourceDiffusableParserServiceTest

EwannRoux committed:
fix: return json with message instead of plaintext in exception handler

EwannRoux committed:
feat: rework filter, cache, DAO and parser

EwannRoux committed:
fix: fix path example in conf

EwannRoux committed:
fix: change log level of logs in CacheServiceJsonImpl

EwannRoux committed:
chore: exclude json files from packaging

EwannRoux committed:
test: create RessourceDiffusable factory for test mocks

EwannRoux committed:
test: adapt RessourceDiffusableParserServiceTest tests to new interface methods

EwannRoux committed:
test: create RessourceDiffusableFilterTest

EwannRoux committed:
test: create CacheServiceJsonImplTest

EwannRoux committed:
test: add required json files for parser tests

EwannRoux committed:
[maven-release-plugin] prepare release ressources-diffusables-api-1.0.0

EwannRoux committed:
[maven-release-plugin] prepare for next development iteration
